tf2_eigen/CMakeLists.txt

```diff
@@ -26,11 +26,12 @@ if(BUILD_TESTING)
   ament_add_gtest(tf2_eigen-test test/tf2_eigen-test.cpp)
   if(TARGET tf2_eigen-test)
     target_include_directories(tf2_eigen-test PUBLIC
-      include
-      ${tf2_INCLUDE_DIRS}
-      ${tf2_ros_INCLUDE_DIRS}
-      ${Eigen_INCLUDE_DIR}
-    )
+      include)
+    set(Eigen)
+    ament_target_dependencies(tf2_eigen-test
+      tf2
+      tf2_ros
+      Eigen)
```

Comment:
Since the package config file of `Eigen3` doesn't set standard compliant variable I would rather not use it with `ament_target_dependencies`, e.g. `Eigen_FOUND` won't be set. Instead I would call:

1. `target_include_directories` for `include`
2.  `ament_target_dependencies` with the two tf2 package names
3.  `target_include_directories` for `EIGEN3_INCLUDE_DIR` (and getting rid of the custom variable `Eigen_INCLUDE_DIR(S)`)